src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -149,18 +149,17 @@ public Container create(String appId, AppDeploymentRequest request, Integer port
 			else {
 				container.addNewPort().withContainerPort(port).endPort();
 			}
-			container.withReadinessProbe(
-				new ProbeCreator(port, properties.getReadinessProbePath(), properties.getReadinessProbeTimeout(),
-					properties.getReadinessProbeDelay(), properties.getReadinessProbePeriod(), "readiness",
-					request.getDeploymentProperties()).create()).withLivenessProbe(
-				new ProbeCreator(port, properties.getLivenessProbePath(), properties.getLivenessProbeTimeout(),
-					properties.getLivenessProbeDelay(), properties.getLivenessProbePeriod(), "liveness",
-					request.getDeploymentProperties()).create());
 		}
 
-		//Add additional specified ports.  Further work is needed to add probe customization for each port.
 		List<Integer> additionalPorts = getContainerPorts(request);
-		if (!additionalPorts.isEmpty()) {
+		if(properties.getLivenessProbePort() != null && !additionalPorts.contains(properties.getLivenessProbePort())) {
```

Comment:
Hi @chrisjs. I figured this would be okay? I am not really sure why this is problematic? I figured because they were probes, whether the ports were already present in the additionalPorts or introduced via the spring.cloud.deployer.kubernetes.*ProbePort property, they still need to be added an exposed to the container?